### PR TITLE
DOC: add examples for special.eval_legendre()

### PR DESCRIPTION
**Pull request by colbertz2**
#### Reference issue
Contributes to gh-7168

#### What does this implement/fix?
Add example code to the docstring for scipy.special.eval_legendre()

### CODE REVIEW

**Comment by AtsushiSakai** [comment]
build doc in CircleCI was failed. Cloud you please merge master branch to your branch, then push your branch again?

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -2951,6 +2951,43 @@ def add_newdoc(name, doc):
...
+    >>> eval_legendre(1, X)
+    array([-1. , -0.5,  0. ,  0.5,  1. ])
+
+    Evaluate Legendre polynomials of order 0 through 5 at x = 0
+
+    >>> N = range(0, 5)
+    >>> eval_legendre(N, 0)
+    array([ 1.   ,  0.   , -0.5  ,  0.   ,  0.375])
+
+    Plot Legendre polynomials of order 0 through 5
```
Comment (rlucas7) [inline scipy/special/add_newdocs.py:2976]: this line reads 0 through 0 but the range is ***not*** inclusive of the second argument.

File: scipy/special/add_newdocs.py [hunk 2]
```diff
@@ -2951,6 +2951,43 @@ def add_newdoc(name, doc):
...
+
+    >>> X = np.linspace(-1, 1)  # Plot over -1 to 1 domain
```
Comment (rlucas7) [inline scipy/special/add_newdocs.py:2978]: safe to omit the comments on these lines

File: scipy/special/add_newdocs.py [hunk 3]
```diff
@@ -2951,6 +2951,43 @@ def add_newdoc(name, doc):
...
+
+    >>> import matplotlib.pyplot as plt
+    >>> for n in range(0, 5):       # Plot each polynomial of order n
```
Comment (rlucas7) [inline scipy/special/add_newdocs.py:2981]: see comment above about exclusivity of second argument in `range`

**Review by rlucas7 — CHANGES_REQUESTED** [review]
@colbertz2 left a couple comments to be addressed.

**Review by rlucas7 — CHANGES_REQUESTED** [review]
@colbertz2 from my end it looks like the changes did not get pushed to the PR. In this PR the docstring reads (one example of 2): 

```
    Evaluate Legendre polynomials of order 0 through 5 at x = 0
    >>> N = range(0, 5)
    >>> eval_legendre(N, 0)
    array([ 1.   ,  0.   , -0.5  ,  0.   ,  0.375])
```

But if you run the `range(0, 5)` in python you'll see: 
```
>>> for i in range(0,5):
...   print(i) 
... 
0
1
2
3
4
>>> 
```
so I recommend to change instances of `N = range(0, 5)` to `N = range(0, 6)` and the rest of the PR can stay the same. If you can also squash the commits (again) via a `git rebase` that would help keep the commit history clean. 

Please let me know if something I'm requesting is unclear. 

EDIT: The `git rebase` is a nice thing to keep the git history clean but not strictly required. A changing of the docstring is though as the numbering is not consistent with the plot the code generates (the 5th Legendre polynomial is not generated). 

EDIT2: For some reason the github ui wasn't showing the changes although the second commit was there, the changes are made. However it seems that the counting of the Legendre polynomials starts at `0` so `0 to 4` as @colbertz2 originally had was correct.

**Review by rlucas7 — APPROVED** [review]
Docstring w/changes now reads  `Legendre polynomials of order 0 through order 4` which ***does*** correspond to the `range(0, 5)` sorry about the confusion with the github ui on this one @colbertz2 

marking approved, squashing and merging.

**Comment by colbertz2** [comment]
Thank you! That does all make sense, and yeah I goofed up range the first time around. 

I can rebase later today to squash those commits. It will be good practice, my experience with rebase is limited and I want to get to know it better

**Pull request merged**